### DISCUSSION TITLE
Automatically load code coverage overview

### DISCUSSION THREAD

**Discussion by morremeyer** [michenriksen/gokiburi#9] [Ideas]
I like to check my code coverage when modifying tests. If gokiburi could automatically refresh the coverage view with new data when tests are run, that would be great!

**Comment by michenriksen** [comment]
I think this would be a great workflow improvement! 

I'm thinking about how the application should behave in a scenario where a new test result does not have coverage data on the currently open file. 

A simple implementation would be to simply ignore any new test results unless it has coverage data on the viewed file. Maybe it could display a little banner on top that tells the user that coverage data has been updated with the latest result.

The UI could also have a little checkbox, either in the settings drawer, or in the coverage view top-bar to toggle automatic refreshing. This could also help a user discover the feature and understand what is going on.

Does this sound good to you?

**Reply by morremeyer** [reply]
That sounds very good to me, yes.

I think ignoring the tests results aka. not refreshing the window/file when there's no data for the open file is what users would expect.

Making it a togglable setting also helps people who explicitly **don't** want this behavior.

Showing an unobtrusive banner sounds like an idea, especially since another sound could be confused for new test results.

**Comment by michenriksen** [comment]
- Issue: #11
